dmoj/judgeenv.py

```diff
@@ -324,10 +349,11 @@ def get_supported_problems_and_mtimes():
     """
     problems = []
     for dir in get_problem_roots():
-        for problem in os.listdir(dir):
-            problem = utf8text(problem)
-            if os.access(os.path.join(dir, problem, 'init.yml'), os.R_OK):
-                problems.append((problem, os.path.getmtime(os.path.join(dir, problem))))
+        if os.path.isdir(dir):
```

Comment:
When can this happen?

Reply:
if you delete a folder that previously contained a problem, it can cause the scan to crash

Reply:
Let's invert this to read `if not ...: continue` and leave a comment.

Is this new behavior, or did it always crash this way and this is fixing a bug?

Reply:
Just tested on the judges we run in prod, the bug exists there too